src/nplinker/metabolomics/IMolecularFamilyLoader.py

```diff
@@ -0,0 +1,11 @@
+from abc import ABC, abstractmethod
+from typing import Iterable
+
+from nplinker.metabolomics.molecular_family import MolecularFamily
+
+
+class IMolecularFamilyLoader(ABC):
+    
+    @abstractmethod
+    def families(self) -> Iterable[MolecularFamily]:
```

Comment:
`typing.Iterable` should be replaced by `coolections.abc.Iterable`, same for other [container types](https://docs.python.org/3.10/library/collections.abc.html#collections-abstract-base-classes).

I guess getting the family using index and getting the number of families are necessary, if that's true, we should use `collections.abc.Sequence` to narrow the type